### PR TITLE
FPO-149: Add ability to destroy existing keys

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

FPO-149

### What?

I have added/removed/altered:

- [x] Added action to destroy a key
- [x] Fixed an issue with routes
- [x] Backfilled missing tests

### Why?

I am doing this because:

- This is required to enable disabling existing keys and to make sure we've got proper coverage to avoid regressions

### CODE REVIEW

File: src/operations/deleteCustomerApiKey.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    const command = new DeleteApiKeyCommand(input)
+
+    await this.apigatewayClient.send(command)
+  }
+
+  private async deleteInDynamoDb (apiKey: CustomerApiKey): Promise<void> {
+    const input: DeleteItemCommandInput = {
+      TableName: 'CustomerApiKeys',
+      Key: {
+        FpoId: { S: apiKey.FpoId },
```
Comment (shjohnson) [inline src/operations/deleteCustomerApiKey.ts:30]: What does the 'S' mean here?

File: spec/controllers/apiKeysControllerSpec.ts [hunk 2]
```diff
@@ -95,6 +95,52 @@ describe('ApiKeyController', () => {
...
+
+      expect(repository.getKey).toHaveBeenCalledWith('fpoId', 'id')
+      expect(repository.updateKey).toHaveBeenCalledWith(apiKey)
+      expect(res.statusCode).toBe(200)
+      expect(res.data).toEqual({
+        CustomerApiKeyId: '',
+        ApiGatewayId: '',
+        Secret: '',
+        Enabled: false,
+        Description: 'description',
```
Comment (shjohnson) [inline spec/controllers/apiKeysControllerSpec.ts:136]: Do we ever need to check the value of this field before the update?

**Review by shjohnson — APPROVED** [review]
Couple questions but otherwise looks good!

File: src/operations/deleteCustomerApiKey.ts (hunk 1 above)
Comment (willfish) [reply src/operations/deleteCustomerApiKey.ts:30]: It's just specifying this is a string and is part of some of the Input schemas.

Comment (shjohnson) [reply src/operations/deleteCustomerApiKey.ts:30]: ok cool thats what i thought 👍 Just wanted to check

File: spec/controllers/apiKeysControllerSpec.ts (hunk 2 above)
Comment (willfish) [reply spec/controllers/apiKeysControllerSpec.ts:136]: No we don't strictly and could just validate a subset of the keys but it seemed cheap enough to just check all the fields :shrug:

**Pull request merged**